app/jobs/socure_docv_results_job.rb

```diff
@@ -42,14 +42,62 @@ def perform(document_capture_session_uuid:, async: true, docv_transaction_token_
           passport_image_fingerprint: nil,
           selfie_image_fingerprint: nil,
         )
+
+        record_attempt(docv_result_response:, doc_pii_response:)
         return
       end
     end
+
+    record_attempt(docv_result_response:, doc_pii_response:)
```

Comment:
Is `doc_pii_response` always defined here? If the response was not successful, it seems like it wouldn't be.

Reply:
hm on further inspection via specs, the `doc_pii_response` is defined as `nil` at this moment, even if the earlier condition doesn't run.

apparently `The local variable is created when the parser encounters the assignment, not when the assignment occurs` ([source](https://ruby-doc.org/3.4.1/syntax/assignment_rdoc.html#label-Local+Variables+and+Methods) -- i don't think i knew this!)